### DISCUSSION TITLE
context switching between schemas, table not found

### DISCUSSION THREAD

**Discussion by arbansal01** [pgbouncer/pgbouncer#672] [Q&A]
Whenever there is context switching between schemas like from public schema to tenant schema or from tenant to public schema then we start getting errors like <table> not found
We have even tried to use set search_path query but no success
We are getting table not found errors with transaction pool mode. And the same scenarios work fine with session pool mode

**Comment by petere** [comment]
By definition, transaction pooling mode will not work correctly if your queries depend on session state. Setting a schema search path is session state. It would work if you set the schema search path inside the same transaction that you run your query from. But you'd have to do that in each transaction.
